ranker/shared/ranker_obj.py

```diff
@@ -150,26 +176,28 @@ def graph_laplacian(self, rgraph, answer):
                         weight_dict[subject_index][object_id][edge_source][edge_property] = val
 
         qedge_qnode_ids = set([frozenset((e['subject'], e['object'])) for e in self.qedge_by_id.values()])
-        laplacian = np.zeros((num_nodes, num_nodes))
+        weighted_graph = np.zeros((num_nodes, num_nodes))
         for subject_index in range(num_nodes):
             q_node_id_subject = node_ids[subject_index][0]
             for object_id in range(num_nodes):
                 q_node_id_object = node_ids[object_id][0]
                 edge_qnode_ids = frozenset((q_node_id_subject, q_node_id_object))
 
-                # Set default weight (or 0 when edge is not a qedge)
-                weight = self.DEFAULT_WEIGHT if edge_qnode_ids in qedge_qnode_ids else 0
-
+                
+                weight = 0
                 for source in weight_dict[subject_index][object_id].keys():
                     for property in weight_dict[subject_index][object_id][source].keys():
                         source_w = weight_dict[subject_index][object_id][source][property]
-                        weight = weight + source_w * source_weight(source, property, source_weights=self.source_weights, unknown_source_weight=self.unknown_source_weight)
-                laplacian[subject_index, object_id] += -weight
-                laplacian[object_id, subject_index] += -weight
-                laplacian[subject_index, subject_index] += weight
-                laplacian[object_id, object_id] += weight
+                        weight = weight + (1-weight)*source_w * source_weight(source, property, source_weights=self.source_weights, unknown_source_weight=self.unknown_source_weight)
```

Comment:
I feel like maybe we should somehow use parallel_combine here. Otherwise the logical in parallel combine will fall out of sync with this if we decide to change it.

Reply:
Changed to using parallel combine.